homeassistant/components/vallox/__init__.py

```diff
@@ -183,6 +196,22 @@ async def async_set_profile_fan_speed_boost(
             return False
         return True
 
+    async def async_set_profile(
+        self, profile: str, duration: int | None = None
+    ) -> bool:
+        """Activate profile for given duration."""
+        _LOGGER.debug("Activating profile %s for %d min", profile, duration)
+        try:
+            await self._client.set_profile(
+                PRESET_MODE_TO_VALLOX_PROFILE[profile], duration
+            )
+        except ValloxApiException as err:
```

Comment:
Let's capture exception here if profile name does not exist in the `PRESET_MODE_TO_VALLOX_PROFILE`

Reply:
Actually I see that it will be blocked by setting in services.yaml